### PR TITLE
Add types to utilities

### LINKED ISSUE #6: Convert JavaScript utility modules to TypeScript
See [`master/api/_utils`](https://github.com/js-hackers/weather-api/tree/master/api/_utils)

### PR DESCRIPTION
**Pull request by jsejcksn**
@samueldlay I started the first file for you.

### CODE REVIEW

File: api/_utils/fetch-weather-data.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import fetch from 'node-fetch';
 
 export type OWMParams = {[key: string]: string};
 
-type WeatherCondtion = {
+type WeatherCondition = {
```
Comment (jsejcksn) [inline api/_utils/fetch-weather-data.ts]: @samueldlay Thanks for catching this spelling mistake

**Comment by jsejcksn** [comment]
@samueldlay LGTM

If you do the squash and merge, be sure to delete everything in the description except the `Co-authored...` line

**Comment by jsejcksn** [comment]
@samueldlay I think it's ready to be merged—just waiting on your review.

**Review by samueldlay — APPROVED** [review]
TypeScript is boss!

**Pull request merged**